mp4parse/src/lib.rs

```diff
@@ -2286,6 +2465,184 @@ fn read_schm<T: Read>(src: &mut BMFFBox<T>) -> Result<SchemeTypeBox> {
+            BoxType::SortArtistEntry => meta.sort_artist = read_string_data(&mut b)?,
+            BoxType::SortAlbumEntry => meta.sort_album = read_string_data(&mut b)?,
+            BoxType::SortAlbumArtistEntry => meta.sort_album_artist = read_string_data(&mut b)?,
+            BoxType::SortComposerEntry => meta.sort_composer = read_string_data(&mut b)?,
+            BoxType::TrackNumberEntry => {
+                if let Some(trkn) = read_u8_data(&mut b)? {
+                    meta.track_number = trkn.get(3).copied();
+                    meta.total_tracks = trkn.get(5).copied();
+                };
+            },
+            BoxType::DiskNumberEntry => {
+                if let Some(disk) = read_u8_data(&mut b)? {
+                    meta.disc_number = disk.get(3).copied();
+                    meta.total_discs = disk.get(5).copied();
+                };
+            },
+            BoxType::TempoEntry => meta.beats_per_minute = read_u8_data(&mut b)?
+                .and_then(|tmpo| tmpo.get(1).copied()),
+            BoxType::CompilationEntry => meta.compilation = read_bool_data(&mut b)?,
+            BoxType::AdvisoryEntry => meta.advisory = read_u8_data(&mut b)?
+                .and_then(|rtng| {
+                    Some(match rtng.get(0)? {
+                        2 => AdvisoryRating::Clean,
+                        0 => AdvisoryRating::Inoffensive,
+                        r => AdvisoryRating::Explicit(*r),
+                    })
+                }),
+            BoxType::MediaTypeEntry => meta.media_type = read_u8_data(&mut b)?
+                .and_then(|stik| {
+                    Some(match stik.get(0)? {
+                        0 => MediaType::Movie,
+                        1 => MediaType::Normal,
+                        2 => MediaType::AudioBook,
+                        5 => MediaType::WhackedBookmark,
+                        6 => MediaType::MusicVideo,
+                        9 => MediaType::ShortFilm,
+                        10 => MediaType::TVShow,
+                        11 => MediaType::Booklet,
+                        s => MediaType::Unknown(*s)
+                    })
+                }),
+            BoxType::PodcastEntry => meta.podcast = read_bool_data(&mut b)?,
+            BoxType::TVSeasonNumberEntry => meta.tv_season = read_u8_data(&mut b)?
+                .and_then(|tvsn| tvsn.get(3).copied()),
+            BoxType::TVEpisodeNumberEntry => meta.tv_episode_number = read_u8_data(&mut b)?
+                .and_then(|tves| tves.get(3).copied()),
+            BoxType::GaplessPlaybackEntry => meta.gapless_playback = read_bool_data(&mut b)?,
+            BoxType::CoverArtEntry => meta.cover_art = read_multiple_u8_data(&mut b).ok(),
+            _ => skip_box_content(&mut b)?,
+
+        };
+        check_parser_state!(b.content);
+    }
+    Ok(())
+}
+
+fn read_bool_data<T: Read>(src: &mut BMFFBox<T>) -> Result<Option<bool>> {
```

Comment:
Rename these new read_* additions to read_ilst_* or move them inside read_ilst to make it clear they're ILST-specific.